mesonbuild/interpreter.py

```diff
@@ -2194,17 +2135,10 @@ def override_find_program_method(self, args, kwargs):
 
     @FeatureNew('meson.override_dependency', '0.54.0')
     @permittedKwargs({'native'})
-    def override_dependency_method(self, args, kwargs):
-        if len(args) != 2:
-            raise InterpreterException('Override needs two arguments')
+    @typed_pos_args('override', str, DependencyHolder)
+    def override_dependency_method(self, args: T.Tuple[str, DependencyHolder], kwargs):
         name = args[0]
-        dep = args[1]
-        if not isinstance(name, str) or not name:
-            raise InterpreterException('First argument must be a string and cannot be empty')
```

Comment:
You still have to check name is not empty, otherwise you could override the special `''` dependency.

Reply:
You're right, I've changed that in the most recent version. The message did change a bit though.